pkg/aggregator/demultiplexer_agent.go

```diff
@@ -0,0 +1,561 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package aggregator
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/aggregator/internal/tags"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/containerlifecycle"
+	"github.com/DataDog/datadog-agent/pkg/epforwarder"
+	"github.com/DataDog/datadog-agent/pkg/forwarder"
+	"github.com/DataDog/datadog-agent/pkg/metrics"
+	"github.com/DataDog/datadog-agent/pkg/serializer"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// DemultiplexerWithAggregator is a Demultiplexer running an Aggregator.
+type DemultiplexerWithAggregator interface {
```

Comment:
Maybe we should call this `AgentDemultiplexerWithAggregator` to be consistent with the new naming - and it would increase clarity that it would be incorrect to try to cast a serverless demultiplexer to this type.

Reply:
I think this already serves the intended purpose:
  * first, it is representing any Demultiplexer with an Aggregator, to be a `DemultiplexerWithAggregator()` you have to provide the `Aggregator()` func (and all methods from the `Demultiplexer` interface ofc).
  * the serverless demultiplexer can't be casted to this since it does not provide the `Aggregator()` method anymore.

It lives in `demultiplexer_agent.go` but you are right that it may be better suited for `demultiplexer.go` in order to be 100% consistent.
WDYT? I may be missing something.